tests/unit/test_network_connections.py

```diff
@@ -3570,6 +3570,32 @@ def test_ipv6_dns_without_ipv6_config(self):
             0,
         )
 
+    def test_ipv6_dns_when_ipv6_method_is_ignore(self):
+        """
+        Test that configuring IPv6 DNS is not allowed when IPv6 method is ignore.
+        """
+        validator = network_lsr.argument_validator.ArgValidator_ListConnections()
+        ipv6_dns_with_ipv6_method_ignore = [
+            {
+                "name": "test_ipv6_dns",
+                "type": "ethernet",
+                "ip": {
+                    "dhcp4": False,
+                    "auto6": False,
+                    "dns": ["2001:db8::20"],
+                },
+            }
+        ]
+
+        self.assertRaisesRegexp(
+            ValidationError,
+            "IPv6 needs to be enabled to support IPv6 nameservers.",
+            validator.validate_connection_one,
+            "nm",
+            validator.validate(ipv6_dns_with_ipv6_method_ignore),
+            0,
+        )
```

Comment:
I should add another test case of which the IPv6 method is manual, test and see if the error is raised again.